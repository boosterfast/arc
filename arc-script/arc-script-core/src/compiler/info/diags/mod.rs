pub mod repr;
pub mod to_codespan;
pub mod sink;
pub(crate) mod utils;

pub use repr::*;
pub use sink::*;
