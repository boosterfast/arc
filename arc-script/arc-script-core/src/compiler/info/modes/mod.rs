pub mod repr;
/// Module for representing sink and source information.
pub mod connector;

pub use repr::*;
