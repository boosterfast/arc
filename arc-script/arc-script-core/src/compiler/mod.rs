/// Module for representing Arc Queries.
pub(crate) mod query;
/// Module for representing the Abstract Syntax Tree.
///
/// NB: This module is public so that external libraries can construct ASTs
/// without having to generate source code.
pub mod ast;
/// Module for representing Dataflow graphs. Exports data types for staging
/// `HIR` functions.
pub mod dfg;
/// Module for representing Higher Order Intermediate Representation code.
pub(crate) mod hir;
/// Module for representing side information.
pub mod info;
/// Module for representing Multi-Level Intermediate Representation code.
pub(crate) mod mlir;
/// Module for representing Arcon code.
pub(crate) mod arcon;

// Module for incremental compilation.
// pub(crate) mod database;
/// Module which assembles compilation-pipeline.
pub mod pipeline;

pub(crate) mod pretty;

pub use pipeline::compile;
