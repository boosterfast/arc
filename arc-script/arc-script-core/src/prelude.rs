//! Exports for users who use the compiler as a library.

pub use crate::compiler;
pub use crate::compiler::ast;
pub use crate::compiler::info::diags;
pub use crate::compiler::info::logger;
pub use crate::compiler::info::modes;

pub use anyhow::Result;
pub use half::bf16;
pub use half::f16;
