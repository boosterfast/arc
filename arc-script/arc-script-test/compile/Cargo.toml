[package]
name        = "arc-script-test-compile"
version     = "0.0.0"
authors     = ["Klas Segeljakt <klasseg@kth.se>"]
edition     = "2018"
license     = "MIT"
description = "Tests to see if arc-scripts compile"

[dependencies]
trybuild        = { version = "1.0.39" }
arc-script-core = { version = "=0.0.0", path = "../../arc-script-core" }
insta           = { version = "1.6.0", features = ["glob"] }

[dev-dependencies]
arcon           = { git = "https://github.com/cda-group/arcon" }
arc-script      = { version = "=0.0.0", path = "../../" }
prost           = { version = "0.7.0" }
