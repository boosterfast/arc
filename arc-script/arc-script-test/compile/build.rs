fn main() {
    arc_script_build::Builder::default()
        .no_exclude(true)
        .source_dirs(["src/tests/expect_pass", "src/tests/expect_mlir_fail_todo"])
        .build();
}
