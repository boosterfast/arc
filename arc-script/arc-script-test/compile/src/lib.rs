#![allow(unused)]

#[cfg(test)]
mod insta;

// #[cfg(test)]
// mod trybuild;

// #[cfg(test)]
// mod compiletest;

mod tests {
    mod expect_pass {
        mod basic_pipe;
        mod binops;
        mod enum_pattern;
        mod enum_pattern_nested;
        mod enums;
        mod fib;
        mod fun;
        mod if_let;
        mod ifs;
        mod lambda;
        mod literals;
        mod nested_if;
        mod option;
        mod path;
        mod pattern;
        mod r#if;
        mod structs;
    }
    mod expect_mlir_fail_todo {
        mod extern_fun;
        mod pipe;
        mod stateful;
        mod task_filter;
        mod task_identity_untagged;
        mod task_map;
        mod task_with_funs;
    }
}
