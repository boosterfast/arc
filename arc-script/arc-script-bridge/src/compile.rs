use arc_script_core::compiler::compile;
use arc_script_core::compiler::info::diags::sink::Buffer;
use arc_script_core::prelude::modes::{Input, Mode, Output};

use proc_macro as pm;
use quote::quote;

use std::fs;
use std::path::PathBuf;

/// arc_script::compile!("script.arc")
///    .stage_fun1(1, 2, 3)
///    .stage_fun2("foo", "bar")
///    .finalize()
pub(crate) fn expand(input: pm::TokenStream) -> pm::TokenStream {
    todo!()
}
