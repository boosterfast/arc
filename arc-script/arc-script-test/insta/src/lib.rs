#![cfg_attr(not(test), allow(dead_code, unused_imports))]

use arc_script_core::prelude::compiler::compile;
use arc_script_core::prelude::diags::Buffer;
use arc_script_core::prelude::modes::{Input, Mode, Output};

use std::env;

fn snapshot(paths: &str, output: Output) {
    env::set_var("INSTA_OUTPUT", "summary");
    env::set_var("INSTA_FORCE_PASS", "1");
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_snapshot_suffix();
    let mut sink = Buffer::no_color();
    settings.bind(|| {
        insta::glob!(paths, |path| {
            println!("Testing {}", path.display());
            let mode = Mode {
                input: Input::File(Some(path.into())),
                output,
                ..Default::default()
            };
            compile(mode, &mut sink).unwrap();
            let s = std::str::from_utf8(sink.as_slice()).unwrap();
            insta::assert_snapshot!(s);
            sink.clear();
        })
    });
}

#[test]
fn test_ast() {
    snapshot(
        "{expect-pass, expect-fail, expect-fail-todo, expect-mlir-fail-todo}/*",
        Output::AST,
    );
}

#[test]
fn test_hir() {
    snapshot(
        "{expect-pass, expect-fail, expect-fail-todo, expect-mlir-fail-todo}/*",
        Output::HIR,
    );
}

#[test]
fn test_dfg() {
    snapshot(
        "{expect-pass, expect-fail, expect-fail-todo, expect-mlir-fail-todo}/*",
        Output::DFG,
    );
}

#[test]
fn test_rust() {
    snapshot(
        "{expect-pass, expect-fail, expect-fail-todo, expect-mlir-fail-todo}/*",
        Output::Rust,
    );
}

#[test]
fn test_mlir() {
    snapshot(
        "{expect-pass, expect-fail, expect-fail-todo}/*",
        Output::MLIR,
    );
}
