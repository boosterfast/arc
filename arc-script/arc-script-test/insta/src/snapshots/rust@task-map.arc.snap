---
source: arc-script-test/insta/src/lib.rs
expression: s
input_file: arc-script-test/insta/src/expect-mlir-fail-todo/task-map.arc
---
use arc_script::arcorn;
use arcon::prelude::*;
#[derive(ArconState)]
pub struct TaskMap {
    #[ephemeral]
    timestamp: Option<u64>,
    #[ephemeral]
    pub f_0: fn(i32) -> i32,
}
impl Operator for TaskMap {
    type IN = Struct5valuei32End;
    type OUT = Struct5valuei32End;
    type TimerState = ArconNever;
    type OperatorState = Self;
    fn handle_element(
        &mut self,
        elem: ArconElement<Self::IN>,
        mut ctx: OperatorContext<Self, impl Backend, impl ComponentDefinition>,
    ) -> OperatorResult<()> {
        self.timestamp = elem.timestamp;
        let event_0 = elem.data.value;
        let (f_0) = (self.f_0.clone());
        ctx.output(ArconElement {
            data: Self::OUT {
                value: f_0(event_0),
            },
            timestamp: self.timestamp,
        });
        Ok(())
    }
    arcon::ignore_timeout!();
    arcon::ignore_persist!();
}
fn Map(f_0: fn(i32) -> i32) -> OperatorBuilder<TaskMap> {
    OperatorBuilder {
        constructor: Arc::new(|b| TaskMap {
            timestamp: None,
            f_0,
        }),
        conf: Default::default(),
    }
}
pub fn x_0(x_1: i32) -> i32 {
    x_1 + 1i32
}
pub fn main(input_0: Stream<Struct5valuei32End>) -> Stream<Struct5valuei32End> {
    let output_0: Stream<Struct5valuei32End> = Stream::operator(input_0, Map(x_0));
    output_0
}
#[arcorn::rewrite]
pub struct Struct5valuei32End {
    value: i32,
}


